conformance/tests/gateway-optional-address-value.yaml

```diff
@@ -0,0 +1,15 @@
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: gateway-without-address-value
+  namespace: gateway-conformance-infra
+spec:
+  gatewayClassName: "{GATEWAY_CLASS_NAME}"
+  addresses:
+  # How should we inject a valid type here?
+  - type: "VALID_ADDRESS_TYPE_PLACEHOLDER"
```

Comment:
This seems like the fundamental challenge of writing a conformance test for this. As far as I know, none of our conformance tests so far require a specific address type. I like your idea of allowing this to be a configurable flag on the test suite and that's probably sufficient as a starting point. It could be useful to expand this in the future to have a separate `SupportedFeature` per type though.